### DISCUSSION TITLE
I'm unable to do absolute navigation to a NavigationPage in Prism 9.0.401-pre

### DISCUSSION THREAD

**Discussion by delaney31** [PrismLibrary/Prism#3096] [.NET MAUI] [answered]
Prior to 9.0.401 we were able to navigate to a MasterPage(FlyOut Page) with a NavigationPage that contained our MainPage like this:
 var result = await _navigationService.NavigateAsync("/MasterPage/NavigationPage/MainPage");
 Debug.WriteLine($"Navigate to MainPage result: {result}");
 
 Now this code is returning a error:  "Page to remove must be contained on this Navigation Page"
 
 I tracked it down to the RemovePage call  in ProcessNavigationForNavigationPage in NavigationService
 
 
        var clearNavigationStack = GetClearNavigationPageNavigationStack(currentPage);
        var isEmptyOfNavigationStack = currentPage.Navigation.NavigationStack.Count == 0;

        List<Page> destroyPages;
        if (clearNavigationStack && !isEmptyOfNavigationStack)
        {
            destroyPages = currentPage.Navigation.NavigationStack.ToList();
            destroyPages.Reverse();

            await currentPage.Navigation.PopToRootAsync(false);
        }
        else
        {
            destroyPages = new List<Page>();
        }

        var topPage = currentPage.Navigation.NavigationStack.LastOrDefault();
        var nextPageType = Registry.GetViewType(UriParsingHelper.GetSegmentName(nextSegment));
        if (topPage?.GetType() == nextPageType)
        {
            if (clearNavigationStack)
                destroyPages.Remove(destroyPages.Last());

            if (segments.Count > 0)
                await UseReverseNavigation(topPage, segments.Dequeue(), segments, parameters, false, animated);

            await DoNavigateAction(topPage, nextSegment, topPage, parameters, onNavigationActionCompleted: (p) =>
            {
                if (nextSegment.Contains(KnownNavigationParameters.SelectedTab))
                {
                    var segmentParams = UriParsingHelper.GetSegmentParameters(nextSegment, parameters);
                    SelectPageTab(topPage, segmentParams);
                }
            });
        }
        else
        {
            await UseReverseNavigation(currentPage, nextSegment, segments, parameters, false, animated);

            if (clearNavigationStack && !isEmptyOfNavigationStack)
            {
                    currentPage.Navigation.RemovePage(topPage);
            }
        }

        foreach (var destroyPage in destroyPages)
        {
            MvvmHelpers.DestroyPage(destroyPage);
        }
    }

**Comment by dansiegel** [comment, marked as answer]
You might check the E2E app for Maui. This is literally a scenario that works there where we do an absolute navigation to a FlyoutPage and from the FlyoutPage.

**Reply by delaney31** [reply]
Thanks. I was able to resolve this using the sample app. The issue was that we were registering the NavigationPage in the Container which I guess you don't need to do anymore.

**Comment by negberts** [comment]
@delaney31 thanks! This helped me out too!
